src/hotspot/share/utilities/intrusiveList.hpp

```diff
@@ -0,0 +1,1508 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#ifndef SHARE_UTILITIES_INTRUSIVELIST_HPP
+#define SHARE_UTILITIES_INTRUSIVELIST_HPP
+
+#include "memory/allStatic.hpp"
+#include "metaprogramming/enableIf.hpp"
+#include "metaprogramming/logical.hpp"
+#include "utilities/align.hpp"
+#include "utilities/debug.hpp"
+#include "utilities/macros.hpp"
+#include <type_traits>
+
+class IntrusiveListEntry;
+class IntrusiveListImpl;
+
+/**
```

Comment:
Most long form comments I see in Hotspot do not use `/** */` and instead uses `//`, can the style be changed to this?

Reply:
I used distinct commenting styles intentionally, in an attempt to distinguish
between "API documentation" and comments on the implementation.  I realize
this is relatively novel in HotSpot code.  I'm open to other ideas, but I
think that distinction is important here.

I don't think it's a good idea to skimp on the API documentation for a utility
like this, leaving questions to be answered by reading the code.  That leads
to people making assumptions based on the current implementation, making it
brittle.  Of course, someone might still inadvertently depend on some aspect
of the current implementation, but if there is clear API documentation then at
least a reviewer can call out violations.

Also, as several people have said to me privately, there's a lot of template
usage here that may not be familiar to all HotSpot devs. I think that makes
requiring someone to read the code to figure out how to use it a problem. I'm
actively striving to avoid requiring someone to read the code simply to use it.